From 2caaedcd9944e42e4ea3a3eeefc867062d9612d3 Mon Sep 17 00:00:00 2001
From: Robert Eckelmann <longnoserob@gmail.com>
Date: Thu, 18 Dec 2025 22:37:45 +0900
Subject: [PATCH 1/2] dts/marvell/mmp3-dell-ariel: import dts from linux-mmp

while linux-mmp project got not updated since 5.8, it still has a
dts for dell-ariel with more hw support.

first try to get this working on mainline again..

Signed-off-by: Robert Eckelmann <longnoserob@gmail.com>
---
 arch/arm/boot/dts/marvell/mmp3-dell-ariel.dts | 331 +++++++++++++++++-
 1 file changed, 321 insertions(+), 10 deletions(-)

diff --git a/arch/arm/boot/dts/marvell/mmp3-dell-ariel.dts b/arch/arm/boot/dts/marvell/mmp3-dell-ariel.dts
index fe6df364a..325d0ccd3 100644
--- a/arch/arm/boot/dts/marvell/mmp3-dell-ariel.dts
+++ b/arch/arm/boot/dts/marvell/mmp3-dell-ariel.dts
@@ -2,7 +2,7 @@
 /*
  * Dell Wyse 3020 a.k.a. "Ariel" a.k.a. Tx0D (T00D, T10D)
  *
- * Copyright (C) 2019 Lubomir Rintel <lkundrak@v3.sk>
+ * Copyright (C) 2019,2020 Lubomir Rintel <lkundrak@v3.sk>
  */
 
 /dts-v1/;
@@ -22,7 +22,8 @@ chosen {
 		#address-cells = <0x1>;
 		#size-cells = <0x1>;
 		ranges;
-		bootargs = "earlyprintk=ttyS2,115200 console=ttyS2,115200";
+		bootargs = "console=tty0 earlyprintk=ttyS2,115200 console=ttyS2,115200 root=/dev/sda3 fbcon=nodefer rootwait lockless_register_fb no_console_suspend";
+		/* bootargs = "console=tty0 earlyprintk=ttyS2,115200 console=ttyS2,115200 root=/dev/sda3 fbcon=nodefer rootwait debug ignore_loglevel drm.debug=65535"; */
 	};
 
 	memory@0 {
@@ -31,20 +32,69 @@ memory@0 {
 		device_type = "memory";
 	};
 
-	ec_input_spi: spi {
-		compatible = "spi-gpio";
+	reserved-memory {
 		#address-cells = <1>;
-		#size-cells = <0>;
+		#size-cells = <1>;
+		ranges;
+
+		display_reserved: framebuffer {
+			compatible = "marvell,mmp2-framebuffer",
+				     "marvell,armada-framebuffer";
+			size = <0x02000000>;
+			alignment = <0x02000000>;
+			no-map;
+		};
+	};
+
+	dvi-connector-0 {
+		compatible = "dvi-connector";
+		ddc-i2c-bus = <&twsi5>;
+		hpd-gpios = <&gpio 62 GPIO_ACTIVE_LOW>;
+		digital;
+		analog;
+
+		port {
+			dvi0_in: endpoint {
+				remote-endpoint = <&encoder_out>;
+			};
+		};
+	};
+
+	dvi-connector-1 {
+		compatible = "dvi-connector";
+		ddc-i2c-bus = <&twsi6>;
+		hpd-gpios = <&gpio 59 GPIO_ACTIVE_LOW>;
+		digital;
+
+		port {
+			dvi1_in: endpoint {
+				remote-endpoint = <&hdmi0_out>;
+			};
+		};
+	};
 
-		num-chipselects = <0>;
-		sck-gpios = <&gpio 55 GPIO_ACTIVE_HIGH>;
-		miso-gpios = <&gpio 57 GPIO_ACTIVE_HIGH>;
-		mosi-gpios = <&gpio 58 GPIO_ACTIVE_HIGH>;
+	sound-card {
+		compatible = "audio-graph-card";
+		label = "Dell Ariel";
+		dais = <&sspa0_dai>;
+		routing = "Headphones", "HPL",
+			  "Headphones", "HPR",
+			  "Speaker External", "SPKLN",
+			  "Speaker External", "SPKLP",
+			  "Speaker External", "SPKRN",
+			  "Speaker External", "SPKRP",
+			  "MIC1N", "Mic Jack",
+			  "MIC1P", "Mic Jack";
+		widgets = "Headphone", "Headphones",
+			  "Speaker", "Speaker External",
+			  "Microphone", "Mic Jack";
 	};
 };
 
 &uart3 {
 	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&uart3_pins>;
 };
 
 &rtc {
@@ -83,6 +133,8 @@ &hsic_phy0 {
 
 &mmc3 {
 	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&mmc3_pins>;
 	max-frequency = <50000000>;
 	status = "okay";
 	bus-width = <8>;
@@ -92,20 +144,76 @@ &mmc3 {
 
 &twsi1 {
 	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&twsi1_pins>;
 
 	rtc@68 {
 		compatible = "dallas,ds1338";
 		reg = <0x68>;
 		status = "okay";
 	};
+
+	regulator@19 {
+		compatible = "marvell,88pg867";
+		reg = <0x19>;
+
+		vcpu: buck1 {
+			regulator-name = "vdd_1v8";
+			regulator-min-microvolt = <1800000>;
+			regulator-max-microvolt = <1800000>;
+			regulator-boot-on;
+			regulator-always-on;
+		};
+	};
 };
 
 &twsi3 {
 	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&twsi3_pins>;
+
+	vga-dvi-encoder@76 {
+		compatible = "chrontel,ch7033";
+		reg = <0x76>;
+
+		ports {
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			port@0 {
+				reg = <0>;
+				encoder_in: endpoint {
+					remote-endpoint = <&lcd0_out>;
+				};
+			};
+
+			port@1 {
+				reg = <1>;
+				encoder_out: endpoint {
+					remote-endpoint = <&dvi0_in>;
+				};
+			};
+		};
+	};
+
+	audio-codec@30 {
+		compatible = "marvell,88ce156";
+		reg = <0x30>;
+
+		port {
+			ce156_0: endpoint {
+				mclk-fs = <256>;
+				clocks = <&audio_clk 0>;
+				remote-endpoint = <&sspa0_0>;
+			};
+		};
+	};
 };
 
 &twsi4 {
 	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&twsi4_pins>;
 
 	embedded-controller@58 {
 		compatible = "dell,wyse-ariel-ec", "ene,kb3930";
@@ -117,8 +225,36 @@ embedded-controller@58 {
 	};
 };
 
+&twsi5 {
+	status = "okay";
+};
+
+&twsi6 {
+	status = "okay";
+};
+
+&twsi6 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&twsi6_pins>;
+};
+
+&lcd0 {
+	status = "okay";
+};
+
+&lcd0_out {
+	remote-endpoint = <&encoder_in>;
+};
+
+&hdmi0_out {
+	remote-endpoint = <&dvi1_in>;
+};
+
 &ssp1 {
 	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&ssp1_pins>;
 	cs-gpios = <&gpio 46 GPIO_ACTIVE_LOW>;
 
 	firmware-flash@0 {
@@ -129,9 +265,11 @@ firmware-flash@0 {
 	};
 };
 
-&ec_input_spi {
+&ssp2 {
 	status = "okay";
 	cs-gpios = <&gpio 56 GPIO_ACTIVE_LOW>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&ssp2_pins>;
 
 	power-button@0 {
 		reg = <0>;
@@ -142,6 +280,33 @@ power-button@0 {
 	};
 };
 
+&asram {
+	status = "okay";
+};
+
+&adma0 {
+	status = "okay";
+};
+
+&audio_clk {
+	status = "okay";
+};
+
+&sspa0 {
+	status = "okay";
+	dmas = <&adma0 0>, <&adma0 1>;
+	dma-names = "tx", "rx";
+
+	sspa0_dai: port {
+		sspa0_0: endpoint {
+			remote-endpoint = <&ce156_0>;
+			frame-master;
+			bitclock-master;
+			dai-format = "i2s";
+		};
+	};
+};
+
 &gpu_2d {
 	status = "okay";
 };
@@ -149,3 +314,149 @@ &gpu_2d {
 &gpu_3d {
 	status = "okay";
 };
+
+&pinctrl {
+/*
+	pinctrl-names = "default";
+	pinctrl-0 = <&my_lovely_pins>;
+
+	my_lovely_pins: pinmux_my_lovely_pins {
+		pinctrl-single,pins =
+			// I2S on GPIO24-GPIO28
+			//        /* SSPA1 (I2S) /
+			//        GPIO25_I2S_BITCLK,
+			//        GPIO26_I2S_SYNC,
+			//        GPIO27_I2S_DATA_OUT,
+			//        GPIO28_I2S_SDATA_IN,
+			<0x0b4 1>, <0x0b8 1>, <0x0bc 1>, <0x0c0 1>, <0x0c4 1>,
+			// I2S on GPIO33-GPIO36
+			<0x0d8 2>, <0x0dc 2>, <0x0e0 2>, <0x0e4 2>,
+			// GPIO on GPIO156-GPIO159 instead of JTAG (doesn't work, stays on 0x000090C0)
+			<0x14c 1>, <0x150 1>, <0x154 1>, <0x158 1>;
+	};
+*/
+
+	uart3_pins: uart3-pins {
+	        pins = "gpio_51", "gpio_52";
+	        function = "uart3";
+	        slew-rate = <2>;
+	        marvell,sleep-output = <0>;
+	        // WKE_0;
+	};
+
+	mmc3_pins: mmc3-pins {
+	        pins = "nd_io10", "nd_io11", "nd_io12", "nd_io13",
+		       "nd_io14", "nd_io15", "nd_io8", "nd_io9",
+		       "sm_ncs0", "sm_ncs1";
+	        function = "mmc3";
+	        slew-rate = <2>;
+	        marvell,sleep-output = <0>;
+	        // WKE_0;
+	};
+
+	twsi1_pins: twsi1-pins {
+		pins = "twsi1_scl", "twsi1_sda";
+		XXX-TWSI;
+		bias-pull-up;
+		slew-rate = <2>;
+		marvell,sleep-input;
+	};
+
+	twsi3_pins: twsi3-pins {
+		pins = "gpio_71", "gpio_72";
+		function = "twsi3";
+		slew-rate = <1>;
+		marvell,sleep-output = <0>;
+		// WKE_0;
+	};
+
+	twsi4_pins: twsi4-pins {
+		pins = "twsi4_scl", "twsi4_sda";
+		function = "twsi4";
+		slew-rate = <1>;
+		marvell,sleep-output = <0>;
+		// WKE_0;
+	};
+
+	twsi5_pins: twsi5-pins {
+		pins = "gpio_41", "gpio_42";
+		function = "twsi5";
+		slew-rate = <1>;
+		marvell,sleep-output = <0>;
+		// WKE_0;
+	};
+
+	twsi6_pins: twsi6-pins {
+		pins = "gpio_47", "gpio_48";
+		function = "twsi6";
+		slew-rate = <1>;
+		marvell,sleep-output = <0>;
+		// WKE_0;
+	};
+
+	lcd_pins: lcd-pins {
+		lcd-fclk-pins {
+			pins = "gpio_74";
+			function = "lcd";
+			slew-rate = <1>;
+			marvell,sleep-output = <0>;
+			// WKE_0;
+		};
+
+		lcd-lclk-pins {
+			pins = "gpio_75";
+			function = "lcd";
+			slew-rate = <1>;
+			marvell,sleep-output = <0>;
+			// WKE_0;
+		};
+
+		lcd-pclk-pins {
+			pins = "gpio_76";
+			function = "lcd";
+			bias-pull-up;
+			slew-rate = <3>;
+			marvell,sleep-output = <0>;
+			// WKE_0;
+		};
+
+		lcd-dena-pins {
+			pins = "gpio_77";
+			function = "lcd";
+			slew-rate = <1>;
+			marvell,sleep-output = <0>;
+			// WKE_0;
+		};
+
+		lcd-data-pins {
+			pins = "gpio_100", "gpio_101", "gpio_78", "gpio_79",
+			       "gpio_80", "gpio_81", "gpio_82", "gpio_83",
+			       "gpio_84", "gpio_85", "gpio_86", "gpio_87",
+			       "gpio_88", "gpio_89", "gpio_90", "gpio_91",
+			       "gpio_92", "gpio_93", "gpio_94", "gpio_95",
+			       "gpio_96", "gpio_97", "gpio_98", "gpio_99";
+			function = "lcd";
+			marvell,sleep-output = <0>;
+			slew-rate = <1>;
+			// WKE_0;
+		};
+	};
+
+	ssp1_pins: ssp1-pins {
+	        pins = "gpio_45", "gpio_44", "gpio_43";
+	        function = "ssp1";
+	        slew-rate = <2>;
+	        marvell,sleep-output = <0>;
+	        bias-pull-up;
+	        // WKE_0;
+	};
+
+	ssp2_pins: ssp2-pins {
+	        pins = "gpio_58", "gpio_57", "gpio_55";
+	        function = "ssp2";
+	        slew-rate = <2>;
+	        marvell,sleep-output = <0>;
+	        bias-pull-up;
+	        // WKE_0;
+	};
+};
-- 
2.52.0

